pkg/frontend/frontend_test.go

```diff
@@ -201,3 +201,51 @@ func TestRoutesAreNamedWithLowerCasePaths(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestAdminJmespathFilter(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    []byte
+		filter   string
+		rootPath string
+		want     []byte
+		wantErr  bool
+	}{
+		{
+			name:  "no jpath",
+			input: []byte("{\"foo\": \"a\"}"),
+			want:  []byte("{\"foo\": \"a\"}"),
+		},
+		{
+			name:   "no root, jpath",
+			filter: "foo",
+			input:  []byte("{\"foo\": \"a\"}"),
+			want:   []byte("\"a\""),
+		},
+		{
+			name:     "root and jpath",
+			filter:   "[0]",
+			rootPath: "foo",
+			input:    []byte("{\"foo\": [\"a\", \"b\"]}"),
+			want:     []byte("{\"foo\":\"a\"}"),
+		},
```

Comment:
Can we modify this case or have an extra case where we have root and have something else apart from root on the top level. I want to make sure that calls like `b, err = adminJmespathFilter(b, jpath, "value")` on objects with pagination will not remove `nextLink` from `b`.

Reply:
Of course, just pushed a modified test case..